### DISCUSSION TITLE
surge dashboard variable $this->rows

### DISCUSSION THREAD

**Discussion by aniltirola** [livewire/livewire#3846] [1. Help] [answered]
I have `surge-master` respository successfully running; After reading different parts of the code I have problems understanding the variable `$this->rows` in `Dashboard.php` | Function `render` codepart:
`return view('livewire.dashboard', [
            'transactions' => $this->rows,
        ]);`

* Where is `$this->rows` initizialized for the first time?
* Where is `$this->rows` filled with data for the first time?

Thanks in advance for guidance

**Comment by jonneroelofs** [comment, marked as answer]
It works a bit different then you expect. It uses computed properties. So $this->rows automatically calls the method getRowsProperty.

https://laravel-livewire.com/docs/2.x/properties#computed-properties

**Reply by aniltirola** [reply]
Thanks, today I learned something new. It seems to be similar like 'defining an accessor' at the laravel docs-page:

https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor
